s4e-backend/src/test/java/pl/cyfronet/s4e/service/PasswordServiceTest.java

```diff
@@ -0,0 +1,246 @@
+        PasswordChangeRequest passwordReset = PasswordChangeRequest.builder()
+                .newPassword(newPassword)
+                .oldPassword(oldPassword)
+                .build();
+
+        assertThat(passwordEncoder.matches(oldPassword, appUserRepository.findByEmail(PROFILE_EMAIL).get().getPassword()), is(true));
+        passwordService.changePassword(passwordReset, PROFILE_EMAIL);
+        assertThat(passwordEncoder.matches(newPassword, appUserRepository.findByEmail(PROFILE_EMAIL).get().getPassword()), is(true));
+    }
+
+    @Test
+    public void shouldntChangePasswordBadRequest() {
+        String newPassword = "password2";
+        String oldPassword = "password";
+        PasswordChangeRequest changeRequest = PasswordChangeRequest.builder()
+                .newPassword(newPassword)
+                .oldPassword(newPassword)
+                .build();
+
+        assertThat(passwordEncoder.matches(oldPassword, appUserRepository.findByEmail(PROFILE_EMAIL).get().getPassword()), is(true));
+        assertThrows(BadRequestException.class, () -> passwordService.changePassword(changeRequest, PROFILE_EMAIL));
+        assertThat(passwordEncoder.matches(newPassword, appUserRepository.findByEmail(PROFILE_EMAIL).get().getPassword()), is(false));
+        assertThat(passwordEncoder.matches(oldPassword, appUserRepository.findByEmail(PROFILE_EMAIL).get().getPassword()), is(true));
+    }
+
+    @Test
+    public void shouldntChangePasswordNotFound() {
+        String newPassword = "password2";
+        String oldPassword = "password";
+        PasswordChangeRequest changeRequest = PasswordChangeRequest.builder()
+                .newPassword(newPassword)
+                .oldPassword(newPassword)
+                .build();
+
+        assertThat(passwordEncoder.matches(oldPassword, appUserRepository.findByEmail(PROFILE_EMAIL).get().getPassword()), is(true));
+        assertThrows(NotFoundException.class, () -> passwordService.changePassword(changeRequest, "email@email.pl"));
+        assertThat(passwordEncoder.matches(newPassword, appUserRepository.findByEmail(PROFILE_EMAIL).get().getPassword()), is(false));
+        assertThat(passwordEncoder.matches(oldPassword, appUserRepository.findByEmail(PROFILE_EMAIL).get().getPassword()), is(true));
+    }
+
+    @Test
+    public void shouldCreatePasswordResetTokenForUser() throws Exception {
+        assertThat(passwordService.findByEmail(PROFILE_EMAIL), isEmpty());
+        passwordService.createPasswordResetTokenForUser( PROFILE_EMAIL);
+        assertThat(passwordService.findByEmail(PROFILE_EMAIL), isPresent());
+    }
+
+    @Test
+    public void shouldntCreatePasswordResetTokenForUser() {
+        assertThat(passwordService.findByEmail(PROFILE_EMAIL), isEmpty());
+        assertThrows(NotFoundException.class, () -> passwordService.createPasswordResetTokenForUser( "mail@email.pl"));
+        assertThat(passwordService.findByEmail(PROFILE_EMAIL), isEmpty());
```

Comment:
You should assert that `findByEmail("mail@email.pl")` return empty, not `findByEmail(PROFILE_EMAIL)`.

Reply:
nice catch :)